keps/sig-autoscaling/5030-attach-limit-autoscaler/README.md

```diff
@@ -0,0 +1,794 @@
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+As part of this KEP we are proposing that, tools like cluster-autoscaler are aware of number of volumes that can be attached to a node.
+
+
+## Motivation
+
+Currently cluster-autoscaler doesn’t take into account, volume-attach limit that a node may have when scaling nodes to support unschedulable pods.
+
+This leads to bunch of problems:
+- If there are unschedulable pods that require more volume than one supported by newly created nodes, there will still be unschedulable pods left.
+
+- Since a node does not come up with a CSI driver typically, usually too many pods get scheduled on a node, which may not be supportable by the node in the first place. This leads to bunch of pods, just stuck.
+
+Once cluster-autoscaler is aware of CSI volume attach limits, we can fix kubernete's builtin scheduler to not schedule pods to nodes that don't have CSI driver installed and if pods require given CSI volumes. Also since 
+cluster-autoscaler isn't aware of CSI volume attach limits, when it scales nodes for pending pods it can't accurately determine how many nodes are required for pending pods that use CSI volumes. For example: if there are 20 pending pods(that use CSI volumes) and assuming cpu, memory and other critireas are met, cluster-autoscaler will not accurately take into account how many nodes are required to satisfy volume attach limits of a node.
+
+After the fixes we are proposing in cluster-autoscaler are made, cluster-autoscaler should accurately calculate number of nodes it needs to spin up to satisfy volume constraints of pending pods. 
+
+### Goals
+
+- Modify cluster-autoscaler so as it is aware of CSI volume limits.
+- Fix scheduler, so as it doesn't schedule pods that require given CSI volume to a node that doesn't have CSI driver installed.
+
+### Non-Goals
+
+- Deschedule pods that can't fit a node because of race conditions.
```

Comment:
Could we call out karpenter integration?

cc @jonathan-innis 

It would be worth at least calling out this integration at some point in the future. I don't think it needs to be solved right now but worth at least seeing if its possible.